tests/CMakeLists.txt

```diff
@@ -26,21 +26,42 @@ if(VEXCL_TEST_COVERAGE)
         )
 endif()
 
-function(add_vexcl_test TEST_NAME TEST_SOURCE)
+# Add one test with backend
+function(_add_vexcl_test TEST_NAME TEST_SOURCE BACKEND)
     add_executable(${TEST_NAME} ${TEST_SOURCE})
 
     if (NOT Boost_USE_STATIC_LIBS)
         target_compile_definitions(${TEST_NAME} PRIVATE BOOST_TEST_DYN_LINK)
     endif ()
 
     target_link_libraries(${TEST_NAME}
-        VexCL::Backend
+        ${BACKEND}
         ${Boost_UNIT_TEST_FRAMEWORK_LIBRARY}
         )
 
     add_test(${TEST_NAME} ${TEST_NAME})
 endfunction()
 
+# Add all requested tests
+function(add_vexcl_test TEST_NAME TEST_SOURCE)
+    if (VEXCL_BACKEND MATCHES "All")
+        if(TARGET VexCL::OpenCL)
+            _add_vexcl_test(${TEST_NAME}_cl ${TEST_SOURCE} VexCL::OpenCL)
+        endif()
+        if(TARGET VexCL::Compute)
+            _add_vexcl_test(${TEST_NAME}_comp ${TEST_SOURCE} VexCL::Compute)
+        endif()
+        if(TARGET VexCL::CUDA)
+            _add_vexcl_test(${TEST_NAME}_cuda ${TEST_SOURCE} VexCL::CUDA)
+        endif()
+        if(TARGET VexCL::JIT)
+            _add_vexcl_test(${TEST_NAME}_jit ${TEST_SOURCE} VexCL::JIT)
+        endif()
+    else()
+        _add_vexcl_test(${TEST_NAME} ${TEST_SOURCE} VexCL::Backend)
+    endif()
+endfunction()
```

Comment:
Why not use `vexcl_add_executables` here (or in `_add_vexcl_test`) that you introduced in #225?

Reply:
Because we need to do an `add_test` on each one anyway.

Reply:
Also was trying to leave the boost unit def line PRIVATE, wasn’t sure if that was needed, mostly familiar with google test and catch.